### PR TITLE
Rename UpgradeManager to ConditionManager

### LINKED ISSUE #354: RA2 is outdated against bleed OpenRA.
I've been struggling for a few hours today to find a good commit in the openra repo (in both the `master` and `bleed` branches) which fully works with the ra2 mod, but there's always some problems.

I've had the most success with some commits around September 2016 where the solution with ra2 will build and run, and the ra2 mod begins to start, indeed it plays the RA2 background music, but there's a bunch of exceptions that happen during type-loading and processing the rules yaml where it fails. I don't want to log bugs because these issues are from targeting much older commits anyway.

The current state of the ra2 project is incompatible with the latest `bleed` commits because the yaml rules have changed considerably and so it needs upgrading, but I'm not familiar with the process, and doubtless the maintainers of this mod are aware of that.

Can the maintainers, or anyone else in the know, advise us which of the most recent openra commits are known to work and run ra2 successfully?

### PR DESCRIPTION
**Pull request by cookgreen**
Update the yaml file for OpenRA bleed. Closes https://github.com/OpenRA/ra2/issues/346. Fixes #354.

### CODE REVIEW

File: rules/animals.yaml [hunk 1]
```diff
@@ -1,5 +1,4 @@
 cow:
-	Inherits: ^Animal
```
Comment (Mailaender) [inline rules/animals.yaml]: You shouldn't remove this.

File: rules/civilians.yaml [hunk 2]
```diff
@@ -1,98 +1,98 @@
 civ1:
-	Inherits: ^CivilianInfantry
+	Inherits: ^CivInfantry
```
Comment (Mailaender) [inline rules/civilians.yaml]: You are introducing new breakages here:
> OpenRA.YamlException: ra2|rules/civilians.yaml:2: Parent type `^CivInfantry` not found

File: rules/soviet-infantry.yaml [hunk 3]
```diff
@@ -1,6 +1,5 @@
 e2:
 	Inherits@1: ^Infantry
-	Inherits@2: ^Parachutable
```
Comment (Mailaender) [inline rules/soviet-infantry.yaml]: You shouldn't remove this, either.

File: rules/allied-infantry.yaml [hunk 4]
```diff
@@ -76,13 +75,11 @@ dog:
 		VoiceSet: DogVoice
 	ProducibleWithLevel:
 		Prerequisites: barracks.infiltrated
-	-TakeCover:
```
Comment (Mailaender) [inline rules/allied-infantry.yaml]: https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Mods.Common/Traits/Infantry/TakeCover.cs is still available and shouldn't be removed for the dog.

**Comment by Mailaender** [comment]
It seems you are having trouble manually editing the rules. Try using the automated utility commands which are explained at https://github.com/OpenRA/OpenRA/wiki/Upgrading-mods-to-a-new-release

**Comment by cookgreen** [comment]
done

**Comment by cookgreen** [comment]
done

**Comment by GraionDilach** [comment]
Pretty sure we have more issues than that. Someone with time (I don't have such atm though) has to go through all yamls and manu8ally update them.

**Comment by pchote** [comment]
Why aren't you guys using the automated upgrade rules?  This is what they're there for.

**Comment by GraionDilach** [comment]
They were used when this was actively maintained. I stepped down from maintainer status when the cake dependency got in and the others apparently fallen out. I actually lost interest so much that I am thinking of reverting https://github.com/GraionDilach/OpenRA.Mods.AS/issues/2 even.

**Comment by cookgreen** [comment]
@pchote i tried to upgrade it, but the upgrade tool crashed(You can check issues #352)

**Comment by drifthat** [comment]
I guess all the yaml files under rules should be converted as well?

**Comment by GraionDilach** [comment]
Yes, a partial "fix" makes no sense and just confuses people.

**Comment by m3ntalsp00n** [comment]
@cookgreen I've got a version that is now working - doesn't crash and runs the same as before as far as I can tell. 

Following from these discussions, there are 'a lot' of yaml changes and the utility upgrade tool can barf all over it if you're not careful. Have to do a manual update, then run the tools to pick up what you've missed. Will need to update that tool for ra2 - but for now just wanna play :)

Would you like me to commit to your repo or just create a new pull request?

**Comment by cookgreen** [comment]
I am glad that if you can commit my repo

**Comment by Mailaender** [comment]
Thanks. I squashed everything together as you introduced some junky merge commits.

**Comment by abcdefg30** [comment]
Thanks for the cleanup! I pushed two minor follow-up commits and will hopefully have some more time to spend on this project this weekend.

**Pull request merged**